ui/e2e/src/fixtures/dashboardTest.ts

```diff
@@ -20,23 +21,101 @@ type DashboardTestOptions = {
 
 type DashboardTestFixtures = {
   dashboardName: string;
+
+  /**
+   * Set to true if the test set will modify the dashboard. When true, the
+   * dashboard will be reset at the end of each test. Do not use in read-only
+   * tests because it will unnecessarily slow them down.
+   */
+  modifiesDashboard: boolean;
   dashboardPage: DashboardPage;
 };
 
+const BACKEND_BASE_URL = 'http://localhost:8080';
+
+async function getDashboardJson(projectName: string, dashboardName: string) {
+  const queryUrl = `${BACKEND_BASE_URL}/api/v1/projects/${projectName}/dashboards/${dashboardName}`;
```

Comment:
perhaps it will worth to reuse the [url-builder](https://github.com/perses/perses/blob/main/ui/app/src/model/url-builder.ts)